.github/workflows/main_push_workflow.yml

```diff
@@ -2,7 +2,7 @@
 name: Main checks after push
 on:
   push:
-    branches: [ '**' ]
+    branches: [ 'main' ]
```

Comment:
So we are not running the pipeline on push to branches anymore?

Reply:
Not until PRs are created. As far as I remember we opened to all branches to build workflows while working on the new implementation. Also, this aligns with other repos: https://github.com/Aiven-Open/s3-connector-for-apache-kafka/blob/main/.github/workflows/main_push_and_pull_request_workflow.yml

Let me know if I'm missing something, though.

Reply:
Ah, yeah, I remembered. But should we actually have 2 workflows or 1 with 2 conditions?

Reply:
We could actually. Was considering keeping them separate to publish images only (TODO) only on the main branch; but could be also done on PRs with a different tag 🤔 

If we prefer this, I'm happy to merge them